src/vivarium/component.py

```diff
@@ -551,6 +573,29 @@ def get_initialization_parameters(self) -> Dict[str, Any]:
             if hasattr(self, parameter_name)
         }
 
+    def create_lookup_tables(self, builder: "Builder") -> None:
+        if (
+            self.name not in builder.configuration
+            or "lookup_tables" not in builder.configuration[self.name]
+        ):
+            return
+        lookup_table_config = builder.configuration[self.name].lookup_tables
+
+        for table_name, table_config in lookup_table_config.items():
+            if "skip_build" in table_config and table_config.skip_build:
+                continue
+            # TODO: make path to configuration the data key when we align artifact
+            # keys with configuration path
+            if table_config.value == "data":
+                table = builder.lookup.build_table(
+                    data=builder.data.load(table_config["key_name"]),
```

Comment:
Is it possible to get a `KeyError` here? If so, do we want to handle that earlier (e.g. in `build_lookup_table_config`?)

Reply:
It depends what you mean. I've set up build_lookup_table_config to have default args to so we do not have KeyErros here. However, we could have an artifact key error and that is tested.